components/feature/push/src/main/java/mozilla/components/feature/push/AutoPushFeature.kt

```diff
@@ -163,25 +157,17 @@ class AutoPushFeature(
      * New encrypted messages received from a supported push messaging service.
      */
     override fun onMessageReceived(message: EncryptedPushMessage) {
-        val type = DeliveryManager.serviceForChannelId(message.channelId)
-        // We may have received a message which specifies `channelId` that we don't know about.
-        // Most likely it was directed at an older version of Firefox running on this device, which still has a valid
-        // FxA device record. Shouldn't be any harm in dropping the message on the floor.
-        if (type == null) {
-            logger.error("Unknown channelID: ${message.channelId}")
-            return
-        }
-        scope.launchAndTry {
-            DeliveryManager.with(connection) {
+        DeliveryManager.with(connection) {
+            coroutineScope.launchAndTry {
                 logger.info("New push message decrypted.")
-                val decrypted = decrypt(
+                val (scope, decryptedMessage) = decryptMessage(
                     channelId = message.channelId,
                     body = message.body,
                     encoding = message.encoding,
                     salt = message.salt,
                     cryptoKey = message.cryptoKey
                 )
-                messageObserverBus.notifyObservers(type, String(decrypted))
+                notifyObservers { onMessageReceived(scope, decryptedMessage) }
```

Comment:
> I also wonder if this could be a source of problems (perf, at least?) if websites decide to programmatically register for a very large number of unique scopes.

We had the MessageBus here for that reason. 🙂 

Do you think we should bring it back?